### DISCUSSION TITLE
Correcting the closed orbit in PyAT

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#324] [Python AT]
Hello everyone,

I am asking if there is a function for  correcting the closed orbit in PyAT by calculating the strength of the required correctors?

I tried to implement that by constructing the response matrix but the results still not good.
I attach a file with my trial.

I'd appreciate your comment on the file
[Correction.zip](https://github.com/atcollab/at/files/7499020/Correction.zip)

**Comment by swhite2401** [comment]
Hello, I could only find a tex file, would you have a python script with your example?

**Reply by elafmusa** [reply]
[Correction.zip](https://github.com/atcollab/at/files/7498972/Correction.zip)


Hello, I attach the file again with the python script.
In my example i tried two methods to invert the response matrix: the pseudo-inverse which leaded to the orbit corrected to the value e-10 and i also tried the SVD method where i think something went wrong as the multiplication of the invers response matrix and the closed orbit gave more than 3 correctors strength values.

Thank you in advance
